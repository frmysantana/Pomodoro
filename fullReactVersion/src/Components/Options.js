import React from 'react';

export default class Options extends React.Component {
    constructor() {
      super();
      this.changeSelected = this.changeSelected.bind(this);
    }
    
    changeSelected(e) {
      this.props.changeSelected(e.target.id);
    }

    // So the user knows that the session timer is selected by default.
    componentDidMount() {
      document.querySelector('#session').checked = true;
    }
    
    render() {
      return (
        <div id="Options">
          <div>
            <label htmlFor="session">Session <br /> Timer</label>
            <input 
              type="radio" name="portion"
              id="session" onClick={this.changeSelected}
            />
          </div>
          
          <div>
            <label htmlFor="break">Break <br /> Timer</label>
            <input type="radio" name="portion" id="break" 
              onClick={this.changeSelected}
            />
          </div>
        </div>
      );
    }
  }
