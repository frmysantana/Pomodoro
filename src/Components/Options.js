import React from 'react';

export default class Options extends React.Component {
    constructor() {
      super();
      this.changeSelected = this.changeSelected.bind(this);
      this.state = {};
    }
    
    changeSelected(e) {
      this.props.changeSelected(e.target.id);
    }
    
    render() {
      return (
        <div>
          <label htmlFor="session">Session Length</label>
          <input 
            type="radio" name="portion"
            id="session" onClick={this.changeSelected}
          />
          
          <label htmlFor="break">Break Length</label>
          <input type="radio" name="portion" id="break" 
            onClick={this.changeSelected}
          />
        </div>
      );
    }
  }