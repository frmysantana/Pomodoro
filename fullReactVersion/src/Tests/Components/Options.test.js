import React from 'react';
import { shallow } from 'enzyme';
import Options from '../../Components/Options';

test('Should render component correctly', () => {
  const wrapper = shallow(<Options />);
  expect(wrapper).toMatchSnapshot();
});