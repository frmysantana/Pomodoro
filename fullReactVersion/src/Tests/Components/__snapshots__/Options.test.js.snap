// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component correctly 1`] = `
<div
  id="Options"
>
  <div>
    <label
      htmlFor="session"
    >
      Session 
      <br />
       Timer
    </label>
    <input
      id="session"
      name="portion"
      onClick={[Function]}
      type="radio"
    />
  </div>
  <div>
    <label
      htmlFor="break"
    >
      Break 
      <br />
       Timer
    </label>
    <input
      id="break"
      name="portion"
      onClick={[Function]}
      type="radio"
    />
  </div>
</div>
`;
